src/editors/playgroundController.ts

```diff
@@ -245,35 +249,23 @@ export default class PlaygroundController {
     );
   }
 
-  async _connectToServiceProvider(): Promise<void> {
-    // Disconnect if already connected.
-    await this._languageServerController.disconnectFromServiceProvider();
-
+  async _activeConnectionChanged(): Promise<void> {
     const dataService = this._connectionController.getActiveDataService();
     const connectionId = this._connectionController.getActiveConnectionId();
+    let mongoClientOption;
 
-    if (!dataService || !connectionId) {
-      this._activeConnectionCodeLensProvider.refresh();
-
-      return;
-    }
-
-    const mongoClientOption =
-      this._connectionController.getMongoClientConnectionOptions();
-
-    if (!mongoClientOption) {
-      this._activeConnectionCodeLensProvider.refresh();
+    this._activeConnectionCodeLensProvider.refresh();
 
-      return;
+    if (dataService && connectionId) {
+      mongoClientOption =
+        this._connectionController.getMongoClientConnectionOptions();
     }
 
-    await this._languageServerController.connectToServiceProvider({
+    await this._languageServerController.activeConnectionChanged({
```

Comment:
Remove the hassle of disconnecting and connecting in the playground and language controller, when we can do it only in the MongoDB service.